src/adapter/base.cr

```diff
@@ -1,10 +1,17 @@
+require "colorize"
 require "../granite_orm"
 require "db"
 
 # The Base Adapter specifies the interface that will be used by the model
 # objects to perform actions against a specific database.  Each adapter needs
 # to implement these methods.
 abstract class Granite::Adapter::Base
+  DATABASE_YML = "config/database.yml"
+  SQL_KEYWORDS_REGEX = %r(SELECT|FROM|WHERE|LIMIT|INNER|VALUES|ORDER|UPDATE|INSERT|DELETE|DROP|
```

Comment:
I like having all of this in one place but now this regexing happens always regardless of wether or not a logger is present. I think we need a performance test for granite. 

It would be a bit more manual but I have been thinking about putting colorization macros into each adapters query methods, that way formatting and colorization can be hand tuned for each type of statement at the adapter level. Then, when wanting to disable the query logger, the macros can be nilled and the code gets stripped at compile time.

Reply:
See https://github.com/crystal-lang/crystal/blob/80a975cf0a98a4ce8464d0069145131667becd70/src/colorize.cr#L95